new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+from.  This avoids a number of potential errors and warnings that can happen in
+the case of conflicts, and gives us control over how we handle conflicts.
+
+Users who wish to set the class of the resource can assign it directly in the
+LWRP with `Namespace::MyResource = self`.
+
+## Deprecations and Errors
+
+Removing `method_missing` means several methods of creating resource names cannot
+be caught.  These are the cases.  We will issue deprecation warnings for each one
+and remove them (and method_missing) come Chef 13.
+
+### Chef::Resource::MyResource = <resource>
+
+When you assign an anonymous class or a resource from another namespace into
+`Chef::Resource`, your new resource will currently be accessible via the DSL.
+Since there is no Ruby method that lets us catch the assignment of a constant,
+there is no way for us to assign an explicit DSL method to handle the resource.
+
+```ruby
+# Deprecated
+Chef::Resource::Blah = Class.new(Chef::Resource)
+```
+
+We will catch this in method_missing, issue a warning letting the user know they
+should use the `provides` directive on their class to get it into the DSL, and remove it come Chef 13.
+
+```ruby
+# Not deprecated (though the assignment becomes unnecessary)
+Chef::Resource::Blah = Class.new(Chef::Resource) { provides :blah }
+```
+
+### Calling `method_missing` directly
+
+If people call `method_missing` directly to invoke DSL methods, and we don't
+define it anymore, it obviously won't actually invoke the methods.  We will
+detect this in method_missing, issue a warning telling people to call the
+actual method instead or use public_send(), and remove it come Chef 13.
+
+### Custom `provides?` without corresponding `provides :name`
+
+If someone creates a resource class with a custom `provides?` method, they can
+currently
+
+```ruby
+class MyResource < Chef::Resource
+  def provides?(node, name)
+    name == :blah
+  end
+end
+```
+
+The implications of this are that we have to scan every single resource when
+you type `blah` in a recipe, just so we can catch this one.  In Chef 12 we will
+issue a warning and tell you to do this instead:
+
+```ruby
+class MyResource < Chef::Resource
+  provides :blah
+end
+```
+
+### Referencing Chef::Resource::MyLWRPResource
+
+When you create an LWRP, it is presently placed in
+`Chef::Resource::CookbooknameResourcename`.  In the spirit of not placing user
+defined stuff into the core Chef namespace, this will no longer happen.  Any
+user that calls `Chef::Resource::MyLwrpResource.new` or extends directly from
+the name `Chef::Resource::MyLwrpResource` will receive a warning.
+
+## Risks To Specific Software
+
+While we're trying to maintain full compatibility, there's risk here, particularly for people doing metaprogramming of resources.  We will run the tests of the following packages, and make sure current versions
+still work (possibly emitting deprecation warnings):
+
+- chef-sugar
+- chefspec
+- chef-rewind
+- Poise
+- Crazytown
```

Comment:
Add Halite to this list, it's spec helper does a lot of this work already as it uses a very large number of anonymous resource and provider classes.